### ISSUE TITLE
Getting authentication failed error message when attempting to publish master branch on private repo

### ISSUE THREAD

**Issue by macu** [desktop/desktop#8623]
### Describe the bug

I have updated GitHub Desktop to the latest version 2.2.3 and signed out and back in under Preferences but I still cannot push my master branch to GitHub. I get the "Authentication failed" message. I own the repository, which is currently empty on GitHub, and it is not archived.

### Version & OS

GitHub Desktop 2.2.3
Mac OS Catalina 10.15.1

### Steps to reproduce the behavior

View my project in the GitHub Desktop app, and press "Publish branch".

### Logs

2019-11-13T01:24:22.198Z - error: [ui] `git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

(The error was parsed as 2: Authentication failed. Some common reasons include: ...

**Comment by tierninho** [comment]
@macu It sounds like you got this error:

![Screen Shot 2019-11-13 at 9 31 56 AM](https://user-images.githubusercontent.com/14828183/68797333-80411a80-05f8-11ea-970a-1395a9e763bf.png)

Try logging in and back in first. Did you attempt any of the above suggestions? Let us know, thanks.

**Comment by macu** [comment]
I tried logging out and back in. I verified the repo is not archived. So the only thing outstanding in those reasons is SSH authentication key, which I'm not sure applies since it is GitHub Desktop.

**Comment by tierninho** [comment]
If you cloned your repo to Desktop via SSH, then you will need SSK keys (see Settings on GitHub.com).

You can check to for existing keys in terminal here: https://help.github.com/en/github/authenticating-to-github/checking-for-existing-ssh-keys

**Comment by macu** [comment]
I didn't clone, I initialized my repo offline, but I added the ssh remote option (git@github.com:...) which required an SSH key to be set up. After doing so I can now push. Thanks.

**Comment by Joseph-Lumpkin** [comment]
For anyone still having a problem, I had to delete my local copy of the repo and re-clone it. I encountered this problem on my linux machine, while my windows machines weren't. I pretty much just backed everything up, minus the .git, deleted the local copy, cloned, then unzipped my backup into my new local copy of the repo. I'm sure it's the same problem @macu encountered. However, I'm still very new to linux (about a week) and like 2 weeks of git sooooo I took the longer, but easier route.
